SpatialGDK/Source/SpatialGDKEditor/Private/SpatialGDKEditor.cpp

```diff
@@ -29,6 +30,63 @@ DEFINE_LOG_CATEGORY(LogSpatialGDKEditor);
 
 #define LOCTEXT_NAMESPACE "FSpatialGDKEditor"
 
+namespace 
+{
+
+bool CheckAutomationToolsUpToDate()
+{
+#if PLATFORM_WINDOWS
+	FString RunUATScriptName = TEXT("RunUAT.bat");
+	FString CmdExe = TEXT("cmd.exe");
+#elif PLATFORM_LINUX
+	FString RunUATScriptName = TEXT("RunUAT.sh");
+	FString CmdExe = TEXT("/bin/bash");
+#else
+	FString RunUATScriptName = TEXT("RunUAT.command");
+	FString CmdExe = TEXT("/bin/sh");
+#endif
+
+	FString UatPath = FPaths::ConvertRelativePathToFull(FPaths::EngineDir() / TEXT("Build/BatchFiles") / RunUATScriptName);
+	
+	if (!FPaths::FileExists(UatPath))
+	{
+		FFormatNamedArguments Arguments;
+		Arguments.Add(TEXT("File"), FText::FromString(UatPath));
+		FMessageDialog::Open(EAppMsgType::Ok, FText::Format(LOCTEXT("RequiredFileNotFoundMessage", "A required file could not be found:\n{File}"), Arguments));
+
+		return false;
+	}
+
+#if PLATFORM_WINDOWS
+	FString FullCommandLine = FString::Printf(TEXT("/c \"\"%s\" -list\""), *UatPath);
+#else
+	FString FullCommandLine = FString::Printf(TEXT("\"%s\" -list"), *UatPath);
+#endif
+
+	FString ListCommandResult;
+	int32 ResultCode = -1;
+	FSpatialGDKServicesModule::ExecuteAndReadOutput(CmdExe, FullCommandLine, FPaths::EngineDir(), ListCommandResult, ResultCode);
```

Comment:
This wrapper should be used primarily for `spatial` since it's a tool not fit for purpose and has strange usages. I'd personally recommend using `FPlatformProcess::ExecProcess`

Reply:
FPlatformProcess does not work in the same way (output is truncated and a cmd window flashes).
What are the downsides of using ExecuteAndReadOutput ? Because otherwise, I'll just inline what it is doing.

Reply:
`ExecuteAndReadOutput` is a wrapper I made around `FPlatformProcess::CreateProc` for spatial but I don't really like it, it obfuscates the options passed into `CreateProc` such as whether to StartHidden etc etc.
The cmd window opens because of the `/c` right?

Reply:
Most likely because the ExecProcess does not use the bHide or bReallyHide options. Both ExecuteAndReadOutput and ExecProcess were passed the same command lines.

Reply:
I highly doubt it, we use `ExecProcess` for `spot` and receive no command window. Could you please test?

Reply:
I did test >.< the difference is that I'm running a batch script through cmd.